src/hooks/useInitialCode.ts

```diff
@@ -19,10 +26,23 @@ const useInitialCode = () => {
     DEFAULT_ADDON_PARAMETERS
   );
 
-  const initialCodeToSet = useMemo(
-    () => (hasValidCode(introCode) ? introCode : DEFAULT_ADDON_STATE.code),
-    [introCode]
-  );
+  const sharedCode = useMemo(() => {
+    const shared = getQueryParam(SNIPPET_SHARE_QUERY_ID);
```

Comment:
basically...yes
we don't do any check on that
so if someone shares a code with us it can contain malicious code
everything in the editor gets executed on the preview

Reply:
a query was sent to appsec
for the meantime let's only accept shared snippets from within the company and be suspicious about playground shares from 3rd parties